### ISSUE TITLE
Unused parameter in function ng_netreg_hdr_build

### ISSUE THREAD

**Issue by mfrey** [RIOT-OS/RIOT#2808]
There seems to be an unused parameter in file ``sys/net/crosslayer/ng_netreg/ng_netreg.c`` in line 133 at 
``` C
133 ng_pktsnip_t *ng_netreg_hdr_build(ng_nettype_t type, ng_pktsnip_t *payload,
134                                   uint8_t *src, uint8_t src_len,
135                                   uint8_t *dst, uint8_t dst_len)
```
This was introduced with 69c593b8. Apparently the unused parameter warning is triggered by the default switch statement while the other case statements use the payload parameter - I don't know if that is actually considered an issue.

**Comment by OlegHahm** [comment]
>  I don't know if that is actually considered an issue.

Warning are always considered an issue.

**Comment by cgundogan** [comment]
I would suggest `(void)`'ing the parameters in the `default` clause

**Comment by OlegHahm** [comment]
:+1:

**Comment by authmillenon** [comment]
The suggested fix was introduced in 716a1bb0 (#2738)

**Comment by OlegHahm** [comment]
Eh, the mentioned fix does not comprise `payload`.

**Comment by authmillenon** [comment]
Fixed in #2809

**Comment by authmillenon** [comment]
Sorry wrong button ^^"
